automation.py

```diff
@@ -2,17 +2,55 @@
 
 # phone_number = r'/^\\(?([0-9]{3})\\)?[-.\\s]?([0-9]{3})[-.\\s]?([0-9]{4})$/'
 # email = r'^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$'
-list =  []
+
 # file = open('potential-contacts.txt','r')
-with open("./potential-contacts.txt")as f:
-  text = f.read()
+with open("potential-contacts.txt")as f:
+  text_file = f.read()
+
 
-print(text)
-# text = file.read()
+
+# print(text)
+#r'\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Z|a-z]{2,}\b'
+# r"(\d{3}[-\.\s]??\d{3}[-\.\s]??\d{4}|\(\d{3}\)\s*\d{3}[-\.\s]??\d{4}|\d{3}[-\.\s]??\d{4})"
 # pattern = r'^[\+]?[(]?[0-9]{3}[)]?[-\s\.]?[0-9]{3}[-\s\.]?[0-9]{4,6}$'
-# pattern = re.compile(r'^[\+]?[(]?[0-9]{3}[)]?[-\s\.]?[0-9]{3}[-\s\.]?[0-9]{4,6}$')
-# matches = pattern.findall(text)
-# for match in matches:
-#     print(match)
+def validate_phone_number():
+  match_list = []
+  new_list = []
+  pattern = re.compile(r"(\d{3}[-\.\s]??\d{3}[-\.\s]??\d{4}|\(\d{3}\)\s*\d{3}[-\.\s]??\d{4}|\d{3}[-\.\s]??\d{4})")
+  matches = pattern.findall(text_file)
+  for match in matches:
+      match_list.append(matches)
+      # print(match_list)
```

Comment:
On 22 you're appending the entire list, instead of the match itself.

Instead, this might work better, and allow you to check for duplicates easier in your next for loop:
```Python
for match in matches:
    match_list.append(match)
```